animeal/src/Modules/Verification/VerificationViewModel.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+
+final class VerificationViewModel: VerificationViewModelLifeCycle, VerificationViewInteraction, VerificationViewState {
+    // MARK: - Private properties
+    private lazy var timeFormatter: DateComponentsFormatter = {
+         let formatter = DateComponentsFormatter()
+         formatter.allowedUnits = [.minute, .second]
+         formatter.unitsStyle = .positional
+         formatter.zeroFormattingBehavior = []
+         return formatter
+     }()
+
+    // MARK: - Dependencies
+    private let model: VerificationModelProtocol
+
+
+    // MARK: - State
+    var onHeaderHasBeenPrepared: ((VerificationViewHeader) -> Void)?
+    var onCodeHasBeenPrepared: ((VerificationViewCode, Bool) -> Void)?
+    var onResendCodeHasBeenPrepared: ((VereficationViewResendCode) -> Void)?
+
+
+    // MARK: - Initialization
+    init(
+        model: VerificationModelProtocol
+    ) {
+        self.model = model
+        setup()
+    }
+
+    // MARK: - Life cycle
+    func setup() {
+        model.fetchNewCodeResponse = { [weak self] modelCode in
```

Comment:
I think, this logic could be moved to VwerificationAssembler.swift

Reply:
No, we could have some private logic here (ex. mapping)

Reply:
How about this solution:

VerificationViewModel.swift
```swift
func fetchNewCodeResponse(code: VerificationModelCode) {
    DispatchQueue.main.async {
        self.onCodeHasBeenPrepared?(
            VerificationViewCode(
                state: VerificationViewCodeState.normal,
                items: code.items.map {
                    VerificationViewCodeItem(identifier: $0.identifier, text: $0.text)
                }
            ),
            true
        )
    }
}
```

VerificationModuleAssembler.swift
```swift
model.fetchNewCodeResponse = { [weak viewModel] modelCode in
    viewModel?. fetchNewCodeResponse(modelCode)
}
```

Reply:
It's not necessary now to do such changes. We have only one action here, if an amount of actions raises, we will divide logic into several methods.